src/activity.ts

```diff
@@ -124,6 +120,49 @@ export function handleAddCollateralActivityEntity(
   activity.save();
 }
 
+export function handleIncreaseDebtActivity(
+  event: IncreaseDebtEvent,
+  controllerId: string
+): void {
+  const token = loadOrCreateERC721Token(event.params.collateralAddress);
+  if (!token) return;
+
+  let activity = Activity.load(event.transaction.hash.toHex());
+  if (!activity) {
```

Comment:
maybe could abstract this stuff to "updateOrCreate" and pass in the new activity title? Or I guess the title depends on if it already exists?